packages/component-library/src/InitialFocus.ts

```diff
@@ -23,14 +40,28 @@ export function InitialFocus({ children }: InitialFocusProps) {
       setTimeout(() => {
         if (node.current) {
           node.current.focus();
-          node.current.setSelectionRange(0, 10000);
+          if (
+            selectText &&
+            (node.current instanceof HTMLInputElement ||
+              node.current instanceof HTMLTextAreaElement)
+          ) {
+            node.current.setSelectionRange(0, 10000);
+          }
         }
       }, 0);
     }
-  }, []);
+  }, [selectText]);
 
   if (typeof children === 'function') {
-    return children(node);
+    return children(node as Ref<HTMLInputElement>);
```

Comment:
No need to cast if type update above is applied
```suggestion
    return children(node);
```